src/main/java/com/adam/springproject/services/UserService.java

```diff
@@ -65,11 +63,37 @@ private boolean isLengthOfPasswordLegal(User user) {
         return !(user.getPassword().length() >= 10 && user.getPassword().length() <= 20);
     }
 
+    public Object signUp(SignUpRequestDto signUpRequestDto) {
+        try {
+            User user = new User(signUpRequestDto);
+            userDao.save(user);
+            return ResponseEntity.status(HttpStatus.CREATED).body("Register successfully");
+        } catch (ConstraintViolationException e) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Information missing");
+        }
+    }
+
     public Object login(LogInRequestDto logInRequestDto) {
-        User user = userDao.findUserByEmail(logInRequestDto.getEmail());
-        if (user == null || !user.getPassword().equals(logInRequestDto.getPassword())) {
+        user = userDao.findUserByEmail(logInRequestDto.getEmail());
+        if (!user.get().getPassword().equals(logInRequestDto.getPassword())) {
+            //登入失敗
+            user = Optional.empty();
             return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("login failed");
         }
-        return ResponseEntity.ok(new LoginResponseDto(user));
+        return ResponseEntity.status(HttpStatus.FORBIDDEN).body("Login successfully");
+    }
+
+    public Object sendMsg(SendMsgRequestDto sendMsgRequestDto) {
+        Date now = new Date();
+        long diff = now.getTime() - lastMessageTime.getTime();
+        if (TimeUnit.SECONDS.convert(diff, TimeUnit.MILLISECONDS) < 5) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).body("Messages are sent every five seconds ");
+        }
+        lastMessageTime = now;
+        if (!user.isPresent()) {
```

Comment:
You can't reuse the user object that you queried in the different requests. You can't assume that the request comes from the same user.